protocol-units/settlement/mcr/client/src/lib.rs

```diff
@@ -4,15 +4,16 @@ use tokio_stream::Stream;
 #[cfg(test)]
 pub mod tests;
 
-pub mod mock;
+#[cfg(feature = "mock")]
+mod mock;
 
 #[cfg(feature = "mock")]
 pub use mock::*;
```

Comment:
... and remove this re-export because then it's less confusing when you are using the mock outside of the crate.

Reply:
Yes I agree, it's what I thought first, but I add a lot of trouble to make everything build in both case mock and eth, so I do the modification, but I came back because I was in trouble. I've to restest now that it's working.

Reply:
In fact, this export is needed. I change it to `pub use mock::McrSettlementClient;`, the same as for the eth feature. We use a feature switch to change the impl of McrSettlementClient.